File: 프로그래머스/lv1/42862. 체육복/체육복.java
import java.util.*;

class Solution {
    public int solution(int n, int[] lost, int[] reserve) {
        int answer = n;
        int[] students = new int[n + 1];
        for(int i = 1 ; i <= n ; i++) students[i]++;
        for(int i : lost) students[i]--;
        for(int i : reserve) students[i]++;
        
        for(int i = 1 ; i <= n ; i++) {
            if(students[i] == 0 && students[i - 1] > 1) {
                students[i] = 1; students[i - 1] -= 1;
            } else if(students[i] == 0 && i < n && students[i + 1] > 1) {
                students[i] = 1; students[i + 1] -= 1;
            } else if(students[i] == 0) {
                answer--;
            }
        }
        return answer;
    }
}

File: 프로그래머스/1/42862. 체육복/체육복.py
def solution(n, lost, reserve):
    save_count = 0
    tmp = list(filter(lambda x : x in reserve, lost))
    for i in tmp:
        lost.remove(i); reserve.remove(i)
    reserve.sort(); lost.sort()
    for i in lost:
        if i - 1 in reserve:
            reserve.remove(i - 1)
            save_count += 1
        elif i + 1 in reserve:
            reserve.remove(i + 1)
            save_count += 1
    return n - len(lost) + save_count